analytical_engine/java/grape-jdk/java-app-runner.sh

```diff
@@ -0,0 +1,215 @@
+    else
+        echo "Using default TEST DIR: /tmp/gstest"
+        TEST_DIR="/tmp/gstest";
+    fi
+fi
+
+echo "Test dir              =${TEST_DIR}"
+# get_test_data
+
+# Parse Java app, figure the fragment Type.
+while read -r line
+do
+  echo ${line}
+  if [[ "${line}" =~ .*"TypeParams".* ]];
+  then 
+    echo "Found type params"
+    delimiter="TypeParams:"
+    #Concatenate the delimiter with the main string
+    string=$line$delimiter
+
+    #Split the text based on the delimiter
+    myarray=()
+    while [[ $string ]]; do
+      myarray+=( "${string%%"$delimiter"*}" )
+      string=${string#*"$delimiter"}
+    done
+
+    #Print the words after the split
+    for value in ${myarray[@]}
+    do
+      echo "value""$value "
+    done
+    typeParams=`echo ${myarray[-1]} | sed 's/ *$//g'`
+    # params=$(echo ${line} | cut -d ':' -f 2)
+    # echo "params: "${params}
+    oidType=$(echo ${typeParams} | cut -d ',' -f 1)
+    vidType=$(echo ${typeParams} | cut -d ',' -f 2)
+    vdataType=$(echo ${typeParams} | cut -d ',' -f 3)
+    edataType=$(echo ${typeParams} | cut -d ',' -f 4)
+    echo "oid:"${oidType}" vid:"${vidType}" vdata: "${vdataType}" edata:"${edataType}
+  fi
+done < <(java -cp ${JAR_PATH} com.alibaba.graphscope.utils.AppBaseParser ${APP})
+
+
+if [[ "${oidType}"x == "java.lang.Long"x ]];
+then
+  echo "OID type: int64"
+else
+  echo "Not a supported OID type"
+  exit 1
+fi
+
+
+if [[ "${vidType}"x == "java.lang.Long"x ]];
+then
+  echo "VID type: int64"
+#  vertex_file_header=${vertex_file_header}",int64_t"
+else
+  echo "Not a supported VID type"
+  exit 1
+fi
+
+if [[ "${vdataType}"x == "java.lang.Long"x ]];
+then
+  vdataType="int64"
+elif [[ "${vdataType}"x == "java.lang.Integer"x ]];
+then
+  vdataType="int"
+elif [[ "${vdataType}"x == "java.lang.Double"x ]];
+then
+  vdataType="double"
+elif [[ "${vdataType}"x == "java.lang.Float"x ]];
+then
+  vdataType="float"
+else
+  echo "Not a supported vdata type"
+  exit 1
+fi
+
+
+if [[ "${edataType}"x == "java.lang.Long"x ]];
+then
+  edataType="int64"
+elif [[ "${edataType}"x == "java.lang.Integer"x ]];
+then 
+  edataType="int"
+elif [[ "${edataType}"x == "java.lang.Double"x ]];
+then
+  edataType="double"
+elif [[ "${edataType}"x == "java.lang.Float"x ]];
+then
+  edataType="float"
+else
+  echo "Not a supported edata type"
+  exit 1
+fi
+
+
+# Add customized header in csv header. #column_types=int64_t|int32_t|double|float
+cmd="import graphscope;from graphscope import JavaApp;"
+cmd=${cmd}"graphscope.set_option(show_log=True);"
+cmd=${cmd}"sess = graphscope.session(cluster_type='hosts');"
+cmd=${cmd}"graph = sess.g(directed=False);"
+cmd=${cmd}"graph = graph.add_vertices(\\\"${TEST_DIR}/property/p2p-31_property_v_0\\\", label=\\\"person\\\", properties=[('weight', '${vdataType}')]);"
+cmd=${cmd}"graph = graph.add_edges(\\\"${TEST_DIR}/property/p2p-31_property_e_0${edge_file_header}\\\", label=\\\"knows\\\", properties=[('dist','${edataType}')]);"
```

Comment:
Could we write this scirpt in python instead? it should be more readable and straightforward.

Making python code in a shell script looks a bit hacky.

Reply:
Yes, I agree.

Reply:
Ok, I will refactor this script as a python script.